app/scripts/modules/core/src/widgets/spinners/Spinner.tsx

```diff
@@ -0,0 +1,61 @@
+import * as React from 'react';
+import autoBindMethods from 'class-autobind-decorator';
+
+const Logo = require('./logo.svg');
+
+export interface ISpinnerProps {
+  size?: 'nano' | 'small' | 'medium' | 'large' | 'page';
+  message?: string;
+  postnote?: string;
+}
+
+@autoBindMethods
+export class Spinner extends React.Component<ISpinnerProps, {}> {
```

Comment:
Compiler used to complain but I don't think it does now. It can just be `React.Component<ISpinnerProps>`, as Adam pointed out.